lib/devise_ldap_authenticatable/ldap/connection.rb

```diff
@@ -55,6 +56,25 @@ def dn
         end
       end
 
+      def domain_user_sid
+        @domain_user_sid ||= begin
+          object_sid_array = binary_sid_to_string(@login_ldap_entry.send(:objectSid)[0]).split('-')
+          object_sid_array.delete_at(-1)
+          primary_group_sid_base = object_sid_array.join('-')
+          primary_group_id = @login_ldap_entry.send(:primaryGroupID)[0]
+          primary_group_sid_base + '-' + primary_group_id
+        end
+      end
+
+      def domain_user_dn(admin_ldap_connection)
+        begin
+          return admin_ldap_connection.search(filter: "objectSid=#{domain_user_sid}")[0][:dn]
+        rescue
```

Comment:
this is very dangerous. you should rescue the things you want, and let the rest pass through. And this is a poor substitute for actually checking if you got an item back.